tests/test_upgrade.py

```diff
@@ -0,0 +1,72 @@
+import pyhf
+import pyhf.schema
+import json
+import logging
+import pytest
+
+
+def test_upgrade_bad_version(datadir):
+    with pytest.raises(ValueError):
+        pyhf.schema.upgrade(to_version='0.9.0')
+
+
+def test_upgrade_to_latest(datadir):
+    ws = json.load(open(datadir.joinpath("workspace_1.0.0.json"), encoding="utf-8"))
```

Comment:
```suggestion
    with open(datadir.joinpath("workspace_1.0.0.json"), encoding="utf-8") as read_file:
        ws = json.load(read_file)
```

Would be good to use context managers for this whole file.